### PR TITLE
Update UI

### LINKED ISSUE #96: Show Team and Task details in UI

### PR DESCRIPTION
**Pull request by Devanshshah1309**
### Changes
1. Disallow selecting teams and tasks through the UI (no change in color)
2. Add links to team description.
3. Add deadline + assignee + status of task to UI
4. Change font (not sure when it reverted back to using the initial one)
5. Allow team names to have spaces (but they cannot begin with a space and cannot be empty)
6. Allow switching of themes via the `theme` command (very hacky way as of now but can't find any better way online either)
7. Keep the detailed team view as large as possible while making sure teams and contacts are reasonably visible

### New UI
Makes more sense to me because it's more likely to have more tasks than links and members. So, we should give more space for tasks.
Screenshot 2022-10-25 at 1 07 59 PM

Fixes #96

### CODE REVIEW

File: src/main/resources/view/LightTheme.css [hunk 1]
```diff
@@ -222,7 +218,7 @@
     -fx-border-width: 2;
     -fx-background-radius: 0;
     -fx-background-color: #F7ECDE;
-    -fx-font-family: "Segoe UI", Helvetica, Arial, sans-serif;
+    -fx-font-family: "Courier New", Helvetica, Arial, sans-serif;
```
Comment (mjgui) [inline src/main/resources/view/LightTheme.css:221]: this should me `monospace` if you want a fallback
```suggestion
    -fx-font-family: "Courier New", monospace;
```
probably the same for the other font things

**Review by mjgui — CHANGES_REQUESTED** [review]
Code looks good!

If possible can we add the following changes?
1. Display the URL also for the link?
2. Also, can we display some message like `URL "https://..." is copied to clipboard!` inside the output box when you click on a link? Now it's not clear what clicking on a link is supposed to do
3. Dark/Light mode
4. Help link theming

File: docs/UserGuide.md [hunk 2]
```diff
@@ -63,7 +63,8 @@ Examples:
 
 ### Set a new team `set_team`
 
-Sets the application to an existing team, i.e., changes the current "working" team to another. Will throw an error if the team does not exist
+Sets the application to an existing team, i.e., changes the current "working" team to another. Will throw an error if
+the team does not exist
```
Comment (Devanshshah1309) [inline docs/UserGuide.md:67]:
```suggestion
Sets the application to an existing team, i.e., changes the current "working" team to another. Will throw an error if the team does not exist
```

File: docs/UserGuide.md [hunk 3]
```diff
@@ -85,7 +86,8 @@ Examples:
 
 ### Adding a new member to the team `add_member`
 
-Add a new team member to the user’s currently selected team. Will throw an error if the member already exists in the team.
+Add a new team member to the user’s currently selected team. Will throw an error if the member already exists in the
+team.
```
Comment (Devanshshah1309) [inline docs/UserGuide.md:90]:
```suggestion
Add a new team member to the user’s currently selected team. Will throw an error if the member already exists in the team.
```

File: docs/UserGuide.md [hunk 4]
```diff
@@ -149,20 +150,22 @@ Format: `assign_task_rand TASK_INDEX`
...
 * `assign_task_rand 1` will assign the first task on the task list to a random team member.
 
-
 ### Set Deadline for task: `set_deadline`
 
-Set a deadline for an existing task. Will display an error message if task is not found in the user’s team. If
-deadline has been set for the task before, a new deadline will be set if command is run again. The deadline must be specified in YYYY-MM-DD HH:MM format.
+Set a deadline for an existing task. Will display an error message if task is not found in the user’s team. If deadline
+has been set for the task before, a new deadline will be set if command is run again. The deadline must be specified in
+YYYY-MM-DD HH:MM format.
```
Comment (Devanshshah1309) [inline docs/UserGuide.md:157]:
```suggestion
Set a deadline for an existing task. Will display an error message if task is not found in the user’s team. If a deadline has been set for the task before, a new deadline will be set if command is run again. The deadline must be specified in YYYY-MM-DD HH:MM format.
```

File: docs/UserGuide.md [hunk 5]
```diff
@@ -149,20 +150,22 @@ Format: `assign_task_rand TASK_INDEX`
...
 
 Format: `set_deadline TASK_INDEX DEADLINE`
 
 Examples:
-* `set_deadline 1 2023-12-25 23:59` will set the deadline for the first task on the task list as 25 December 2023 11.59pm.
+* `set_deadline 1 2023-12-25 23:59` will set the deadline for the first task on the task list as 25 December 2023
+  11.59pm.
```
Comment (Devanshshah1309) [inline docs/UserGuide.md:163]:
```suggestion
* `set_deadline 1 2023-12-25 23:59` will set the deadline for the first task on the task list as 25 December 2023 11.59pm.
```

File: docs/UserGuide.md [hunk 6]
```diff
@@ -149,20 +150,22 @@ Format: `assign_task_rand TASK_INDEX`
...
 
 ### Delete task from team : `delete_task`
 
-Delete an existing task from the team given the task's index. Will display an error message if the task is not found in the user’s team, i.e., when the index exceeds the number of tasks.
+Delete an existing task from the team given the task's index. Will display an error message if the task is not found in
+the user’s team, i.e., when the index exceeds the number of tasks.
```
Comment (Devanshshah1309) [inline docs/UserGuide.md:168]:
```suggestion
Delete an existing task from the team given the task's index. Will display an error message if the task is not found in the user’s team, i.e., when the index exceeds the number of tasks.
```

File: docs/UserGuide.md [hunk 7]
```diff
@@ -172,8 +175,8 @@ Examples:
 
 ### List tasks in team: `list_tasks`
 
-View all the tasks currently in the user’s team in the form of a list. List can also be filtered based on complete
-or remaining tasks.
+View all the tasks currently in the user’s team in the form of a list. List can also be filtered based on complete or
+remaining tasks.
```
Comment (Devanshshah1309) [inline docs/UserGuide.md:179]:
```suggestion
View all the tasks currently in the user’s team in the form of a list. List can also be filtered based on complete or remaining tasks.
```

File: src/main/java/seedu/address/model/team/Team.java [hunk 8]
```diff
@@ -18,17 +18,24 @@
...
-    public static final String MESSAGE_CONSTRAINTS = "Team names should be alphanumeric";
-    public static final String VALIDATION_REGEX = "\\p{Alnum}+";
+    public static final String MESSAGE_CONSTRAINTS =
+        "Team names should not be blank and cannot begin with a whitespace";
+
+    /*
+     * The first character of the team name must not be a whitespace,
+     * otherwise " " (a blank string) becomes a valid input.
+     */
+    public static final String VALIDATION_REGEX = "[^\\s].*";
```
Comment (marcuspang) [inline src/main/java/seedu/address/model/team/Team.java:28]: If we don’t use alphanumeric strings, we may run into the case where they are trying to escape the string, for e.g. `\` and `”` where they may not want to include it

File: src/main/java/seedu/address/ui/MainWindow.java [hunk 9]
```diff
@@ -180,12 +182,50 @@ void show() {
...
+                oldStyleSheets.add(styleSheet);
+                String newUrl = styleSheet.replaceAll("LightTheme", "DarkTheme");
+                newStyleSheets.add(newUrl);
+            } else if (Pattern.matches(".*DarkTheme.*", styleSheet)) {
+                oldStyleSheets.add(styleSheet);
+                String newUrl = styleSheet.replaceAll("DarkTheme", "LightTheme");
+                newStyleSheets.add(newUrl);
+            }
+        }
+        primaryStage.getScene().getStylesheets().addAll(newStyleSheets);
```
Comment (marcuspang) [inline src/main/java/seedu/address/ui/MainWindow.java:208]: Might be a cause of concern w.r.t. to Law of Demeter here

File: src/main/java/seedu/address/model/team/Team.java (hunk 8 above)
Comment (Devanshshah1309) [reply src/main/java/seedu/address/model/team/Team.java:28]: Hmm okay i'm fine with having alphanumeric strings but spaces should definitely be allowed.. what do you think?

File: src/main/java/seedu/address/ui/MainWindow.java (hunk 9 above)
Comment (Devanshshah1309) [reply src/main/java/seedu/address/ui/MainWindow.java:208]: Yeah it's pretty hacky but I don't think there's a better way to do this.

**Review by marcuspang — CHANGES_REQUESTED** [review]
Everything looks good, except some minor nits. 

For the fonts, I feel that the headings can remain as monospace bolded. However, for description-type fields, they should not be monospace as they are not bolded. Making them monospace decreases the readability especially for dark backgrounds.

**Review by marcuspang — APPROVED** [review]
LGTM

**Pull request merged**